### PR TITLE
First draft of ARCHITECTURE.md

### PR DESCRIPTION
**Pull request by giannissc**

### CODE REVIEW

File: ARCHITECTURE.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# ARCHITECTURE
+Xilem is a frameworks that aims to provide a performant and productive option for Rust GUI.
```
Comment (waywardmonkeys) [inline ARCHITECTURE.md:2]: "framework"

File: ARCHITECTURE.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+This crates evolved from various experiments in the linebender community (Druid, masonry, idiopath, lasagna, crochet, etc)
```
Comment (waywardmonkeys) [inline ARCHITECTURE.md:4]: "Linebender" (not sure about the others).

File: ARCHITECTURE.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+- [Announcing Masonry 0.1, and my vision for Rust UI (2023)](https://poignardazur.github.io/2023/02/02/masonry-01-and-my-vision-for-rust-ui/)
+- [So you want to write a GUI framework (2021)](https://www.cmyr.net/blog/gui-framework-ingredients.html)
+- [Rust GUI Infrastructure (2021)](https://www.cmyr.net/blog/rust-gui-infra.html)
+- [Towards a unified theory of reactive UI (2019)](https://raphlinus.github.io/ui/druid/2019/11/22/reactive-ui.html)
+- [Towards principled reactive UI (2020)](https://raphlinus.github.io/rust/druid/2020/09/25/principled-reactive-ui.html)
+- [Xilem: an architecture for UI in Rust (2022)](https://raphlinus.github.io/rust/gui/2022/05/07/ui-architecture.html)
+- [Advice for the next dozen Rust GUIs (2022)](https://raphlinus.github.io/rust/gui/2022/07/15/next-dozen-guis.html)
+
+## High-level Goals
+- **High Performance.** Lightweight state management, retained widget layer, massively parallel rendering backend (vello), fast startup time, small binary size
```
Comment (waywardmonkeys) [inline ARCHITECTURE.md:24]: "Vello"

File: ARCHITECTURE.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+fn main() {
+    let data = AppData {
+      count: 0
+    }
+    let app = Xilem::new(data, app_logic);
+    app.run_windowed("Application Example".into()).unwrap();
+}
+```
+The most imporant construct at the reactive layer is the `View` trait. `View` implementations need to assign the associated  `State` and `Element` and implement `build()`, `rebuild()` and `message()`. The `build()` is run only once (first run) to initialize the associated `State` create the associated `Element`of the view. The `rebuild()` is used to update the associated `State` and `Element` and it is what enables Xilem's reactivity. It is worth noting three special types of views the `Adapt`, `Memoize` and `AnyView` views.
```
Comment (waywardmonkeys) [inline ARCHITECTURE.md:68]: "important"

Comment (waywardmonkeys) [reply ARCHITECTURE.md:68]: and there are 2 spaces between "associated" and "`State`"

**Review by waywardmonkeys — CHANGES_REQUESTED** [review]
I'd suggest checking for other similar issues ... I just found them via visual scan, not intense search.

**Comment by giannissc** [comment]
I have created this a while ago and updated it a bit today. If its urgent to get into Xilem don't rely on me finding the time to finish it as I will  be packed until the end of the month..!

**Review by waywardmonkeys — APPROVED** [review]
Since this is just a step on a long road ... let's start it.

**Comment by giannissc** [comment]
That's what I thought! Anything else you would like me to add before we merge? @PoignardAzur

**Comment by PoignardAzur** [comment]
LGTM.

**Pull request merged**